lib/livebook_web/live/home_live.ex

```diff
@@ -28,64 +29,65 @@ defmodule LivebookWeb.HomeLive do
-                phx-click="new">
+              <%= live_patch "Import",
+                    to: Routes.home_path(@socket, :import, "url"),
+                    class: "button button-outlined-gray whitespace-nowrap" %>
+              <button class="button button-blue" phx-click="new">
                 New notebook
               </button>
             </div>
           </div>
+
           <div class="h-80">
             <%= live_component LivebookWeb.PathSelectComponent,
-              id: "path_select",
-              path: @path,
-              extnames: [LiveMarkdown.extension()],
-              running_paths: paths(@session_summaries),
-              phx_target: nil,
-              phx_submit: nil do %>
+                  id: "path_select",
+                  path: @path,
+                  extnames: [LiveMarkdown.extension()],
+                  running_paths: paths(@session_summaries),
+                  phx_target: nil,
+                  phx_submit: nil do %>
               <div class="flex justify-end space-x-2">
-                <%= content_tag :button,
-                  class: "button button-outlined-gray whitespace-nowrap",
-                  phx_click: "fork",
-                  disabled: not path_forkable?(@path) do %>
-                  <%= remix_icon("git-branch-line", class: "align-middle mr-1") %>
+                <button class="button button-outlined-gray whitespace-nowrap"
+                  phx-click="fork"
+                  disabled={not path_forkable?(@path)}>
+                  <.remix_icon icon="git-branch-line" class="align-middle mr-1" />
                   <span>Fork</span>
-                <% end %>
+                </button>
                 <%= if path_running?(@path, @session_summaries) do %>
-                  <%= live_redirect "Join session", to: Routes.session_path(@socket, :page, session_id_by_path(@path, @session_summaries)),
-                    class: "button button-blue" %>
+                  <%= live_redirect "Join session",
+                        to: Routes.session_path(@socket, :page, session_id_by_path(@path, @session_summaries)),
+                        class: "button button-blue" %>
                 <% else %>
-                  <%= tag :span, if(File.regular?(@path) and not file_writable?(@path),
-                        do: [class: "tooltip top", aria_label: "This file is write-protected, please fork instead"],
-                        else: []
-                      ) %>
-                    <%= content_tag :button, "Open",
-                      class: "button button-blue",
-                      phx_click: "open",
-                      disabled: not path_openable?(@path, @session_summaries) %>
+                  <span {if(
+                    File.regular?(@path) and not file_writable?(@path),
+                    do: [class: "tooltip top", aria_label: "This file is write-protected, please fork instead"],
+                    else: []
+                  )}>
```

Comment:
I would totally move this to a function. Like `path_span_attributes(@path)`.

Reply:
I took this as an opportunity to rewrite tooltip into component:

```
<.tooltip label="This file is write-protected, please fork instead"
  if={File.regular?(@path) and not file_writable?(@path)}>
  <button class="button button-blue"
    phx-click="open"
    disabled={not path_openable?(@path, @session_summaries)}>
    Open
  </button>
</.tooltip>
```

Let me know if the implementation is alright in terms of diffing :)

Reply:
I think it is fine but I am not sure, I would check over the wire. :)

Reply:
Hmm, after some investigation this is not diff friendly, and I'm not sure it realistically can.

There are several concerns, but the main issue is that we use `assign_new` and `assign` for extra attributes, which means the `<span>` attributes are re-sent whenever the `tooltip` component is rendered **and** we use inner content which may trigger re-render even though other component assigns don't change. In this specific case the button `disabled` prop is re-evaluated on every render, which consequently sends the tooltip attributes over the wire all the time. The dynamic `if` adds on top of that, but it's rather specific to this case.

Due to the `assign` and `assign_new`, even this would re-send the span attributes:

```
<.tooltip label="Some static content">
  <%= @changing %>
</.tooltip>
```

FWIW the tooltip attributes is not that much content as compared to the content that gets send on usual re-render, but we should probably revert to the static spans, unless you have some other ideas

Reply:
I think it makes sense. Before you were gating only the attribute before the conditional, but now you are gating everything, which means that more will change. I am ok with keeping it as a span and then having a span_path_attributes or similar.